src/Renci.SshNet/SftpClient.cs

```diff
@@ -640,6 +758,37 @@ public void DownloadFile(string path, Stream output, Action<ulong> downloadCallb
             InternalDownloadFile(path, output, null, downloadCallback);
         }
 
+#if FEATURE_TAP
+        /// <summary>
+        /// Asynchronously downloads remote file specified by the path into the stream.
+        /// </summary>
+        /// <param name="path">File to download.</param>
+        /// <param name="output">Stream to write the file into.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to observe.</param>
+        /// <returns>A <see cref="Task"/> that represents the asynchronous download operation.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="path" /> is <b>null</b> -or- <paramref name="output" /> is <b>null</b>.</exception>
+        /// <exception cref="SshConnectionException">Client is not connected.</exception>
+        /// <exception cref="SftpPermissionDeniedException">Permission to perform the operation was denied by the remote host. <para>-or-</para> A SSH command was denied by the server.</exception>
+        /// <exception cref="SftpPathNotFoundException"><paramref name="path"/> was not found on the remote host.</exception>/// 
+        /// <exception cref="SshException">A SSH error where <see cref="Exception.Message" /> is the message from the remote host.</exception>
+        /// <exception cref="ObjectDisposedException">The method was called after the client was disposed.</exception>
+        public async Task DownloadFileAsync(string path, Stream output, CancellationToken cancellationToken)
```

Comment:
I propose to leave out this method. See my comment on `UploadFileAsync(...)` for more info.